ResearchKit/Common/ORKKeychainWrapper.m

```diff
@@ -0,0 +1,295 @@
+ 
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice, 
+ this list of conditions and the following disclaimer in the documentation and/or 
+ other materials provided with the distribution. 
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors 
+ may be used to endorse or promote products derived from this software without 
+ specific prior written permission. No license is granted to the trademarks of 
+ the copyright holders even if such marks are included in this software. 
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+
+#import "ORKKeychainWrapper.h"
+
+
+static NSString *_defaultService;
+
+@implementation ORKKeychainWrapper
+
+#pragma mark - Public Methods
+
++ (BOOL)setObject:(id<NSSecureCoding>)object
+           forKey:(NSString *)key
+            error:(NSError **)error {
+    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:object];
+    return [self setData:data
+                  forKey:key
+                 service:[self defaultService]
+             accessGroup:nil
+                   error:error];
+}
+
++ (id<NSSecureCoding>)objectForKey:(NSString *)key
+             error:(NSError *__autoreleasing *)error {
+    NSData *data = [self dataForKey:key
+                            service:[self defaultService]
+                        accessGroup:nil
+                              error:error];
+    return data ? [NSKeyedUnarchiver unarchiveObjectWithData:data] : nil;
+}
+
++ (BOOL)removeObjectForKey:(NSString *)key
+                     error:(NSError **)error {
+    return [self removeItemForKey:key
+                          service:[self defaultService]
+                      accessGroup:nil
+                            error:error];
+}
+
++ (BOOL)resetKeychainWithError:(NSError **)error {
+    return [self removeAllItemsForService:[self defaultService]
+                              accessGroup:nil
+                                    error:error];
+}
+
+#pragma mark - Private Methods
+
++ (NSString *)defaultService {
+    if (!_defaultService) {
```

Comment:
It would be more clean to declare `defaultService` as a local static var.
For example:

```
static NSDateFormatter *__formatter = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        __formatter = [[NSDateFormatter alloc] init];
        [__formatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ssZ"];
        [__formatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_US_POSIX"]];
    });
```